spring-integration-core/src/main/java/org/springframework/integration/channel/MessagePublishingErrorHandler.java

```diff
@@ -70,9 +73,24 @@ public void setDefaultErrorChannel(MessageChannel defaultErrorChannel) {
 	 * @since 4.3
 	 */
 	public MessageChannel getDefaultErrorChannel() {
+		String defaultErrorChannelName = this.defaultErrorChannelName;
+		if (defaultErrorChannelName != null) {
+			if (this.channelResolver != null) {
```

Comment:
I thought about this style before too, but anyway I ended up with the `synchronized` because I was afraid that it would be an overhead to let several threads to call the same bean from `BeanFactory` concurrently. But maybe I am paranoid and it safe to walk to the `BeanFactory` rather than wait on the `synchronized`?..

I can come back to the `synchronized` for consistency or redo other cases but not in this PR.
On demand would sound better.